File: 1.cuenta-bancaria/main.js
class CuentaBancaria {
  // Atributo privado usando la sintaxis #
  #saldo;

  constructor(saldoInicial = 0) {
    this.#saldo = saldoInicial >= 0 ? saldoInicial : 0;
  }

  depositar(monto) {
    if (monto > 0) {
      this.#saldo += monto;
      console.log(`Depósito exitoso: +$${monto.toFixed(2)}`);
    } else {
      console.log("El monto a depositar debe ser mayor a 0.");
    }
  }

  retirar(monto) {
    if (monto > 0 && monto <= this.#saldo) {
      this.#saldo -= monto;
      console.log(`Retiro exitoso: -$${monto.toFixed(2)}`);
    } else {
      console.log("Retiro rechazado: Fondos insuficientes o monto inválido.");
    }
  }

  obtenerSaldo() {
    return this.#saldo;
  }
}

// Prueba de la clase
const miCuenta = new CuentaBancaria(100);
miCuenta.depositar(50);
miCuenta.retirar(30);
miCuenta.retirar(200);
console.log(`Saldo final: $${miCuenta.obtenerSaldo().toFixed(2)}`);

File: 1.cuenta-bancaria/main.py
class CuentaBancaria:
    def __init__(self, saldo_inicial: float = 0.0):
        # Atributo privado (Encapsulamiento)
        self.__saldo = saldo_inicial if saldo_inicial >= 0 else 0.0

    def depositar(self, monto: float) -> None:
        if monto > 0:
            self.__saldo += monto
            print(f"Depósito exitoso: +${monto:.2f}")
        else:
            print("El monto a depositar debe ser mayor a 0.")

    def retirar(self, monto: float) -> None:
        if 0 < monto <= self.__saldo:
            self.__saldo -= monto
            print(f"Retiro exitoso: -${monto:.2f}")
        else:
            print("Retiro rechazado: Fondos insuficientes o monto inválido.")

    def obtener_saldo(self) -> float:
        return self.__saldo


# Prueba de la clase
mi_cuenta = CuentaBancaria(100.0)
mi_cuenta.depositar(50.0)
mi_cuenta.retirar(30.0)
mi_cuenta.retirar(200.0)
print(f"Saldo final: ${mi_cuenta.obtener_saldo():.2f}")